src/main/scala/com/exini/dicom/streams/ParseFlow.scala

```diff
@@ -210,45 +216,62 @@ class ParseFlow private (chunkSize: Int) extends ByteStringParser[DicomPart] {
       }
     }
 
-    case class InDatasetHeader(state: DatasetHeaderState) extends DicomParseStep {
-      private def readDatasetHeader(reader: ByteReader, state: DatasetHeaderState): Option[DicomPart] = {
+    case class InDatasetHeader(datasetHeaderState: DatasetHeaderState) extends DicomParseStep {
+      def maybeSwitchTs(reader: ByteReader, state: DatasetHeaderState): DatasetHeaderState = {
+        reader.ensure(8)
+        val data       = reader.remainingData.take(8)
+        val tag        = bytesToTag(data, state.bigEndian)
+        val explicitVR = Try(VR.withValue(bytesToVR(data.drop(4)))).getOrElse(null)
+        if (isSpecial(tag))
```

Comment:
I wouldn't mind a more descriptive name of the isSpecial-method, e.g.  isSeqItemDelimiter or similar